lib/benchee/benchmark/runner.ex

```diff
@@ -190,7 +239,7 @@ defmodule Benchee.Benchmark.Runner do
          _collector,
          measurements
        )
-       when current_time > end_time do
+       when current_time > end_time and measurements != [] do
```

Comment:
why do we need the additional check?

It'd lead to at least one measurement being taken, right? I guess that's fine but rather implicit (because this doeasn't hit we hit the real benchmark and it does it)

(I was confused as I assumed it was because `Enum.reverse` would error on empty list which didn't make sense

Reply:
We don't have the same initial check for dealing with very fast functions (and the n_times check) for reductions (or memory for that matter) that we do for runtime (and clocks are weird/unreliable), so without this we were seeing cases where we finished but didn't have any reduction counts.